### ISSUE TITLE
Suggestion: typing timeout

### ISSUE THREAD

**Issue by lcrippa** [albertlauncher/python#9]
When an extension (such as wikipedia or google translate) is triggered, for each change in user input a query to the api site is made. This has two drawbacks: it slows down the overall performance of the extension, and it can cause problems for sites with limtations (e.g. github's 30 queries per minute even for authenticated accounts). Maybe it would be useful to insert sort of a "typing timeout" in the python albert library, triggering the extension only after the user input hasn't changed for a time greater than a threshold.

**Comment by ManuelSchneid3r** [comment]
well thats heavily user depending. Some have a rate of 200 others 10. Further it adds a lag. 

> it slows down the overall performance of the extension

thats not true. the queries are multi threaded.

The idea is good, for rate limited services. Wiki does not have a limit afaik. Github is not implemented atm

**Comment by ManuelSchneid3r** [comment]
thats not true. the queries are multi threaded. just realized no they are not (really) due to the GIL. But I hope the gil is released when python urllib waits for responses

**Comment by lcrippa** [comment]
(yes, I tweaked the Wiki extension to apply to github, that's why I bumped into the problem. As a quick and dirty fix I inserted an ending character, such as a dot., to trigger the query, whereas if it's missing the extensiion behaves like websearch.)
> thats not true. the queries are multi threaded.
In that case I'm sorry, I was basing on the output of a debug "print" statement, which showed a query for each partial input.
As for the user-dependent timeout yes, that's right. I was wondering if a low enough threshold would do the job.
